kani-driver/src/concrete_playback.rs

```diff
@@ -11,13 +11,44 @@ use anyhow::{Context, Result};
 use concrete_vals_extractor::{extract_harness_values, ConcreteVal};
 use kani_metadata::HarnessMetadata;
 use std::collections::hash_map::DefaultHasher;
+use std::env;
 use std::ffi::OsString;
-use std::fs::{self, File};
+use std::fs::{self, remove_file, File};
 use std::hash::{Hash, Hasher};
-use std::io::{Read, Write};
-use std::path::Path;
+use std::io::{BufRead, BufReader, BufWriter, Error, Write};
+use std::path::{Path, PathBuf};
 use std::process::Command;
 
+struct TempFile {
+    file: File,
+    temp_path: PathBuf,
+}
+
+impl TempFile {
+    fn new() -> Result<Self, Error> {
+        // Create temp file
+        let mut temp_path = env::temp_dir();
+        temp_path.push("concrete_overwrite.tmp");
+        let temp_file = match File::create(&temp_path) {
+            Ok(f) => f,
+            Err(e) => return Err(e),
+        };
+        Ok(Self { file: temp_file, temp_path })
+    }
+}
+
+impl Drop for TempFile {
+    fn drop(&mut self) {
+        if let Err(e) = self.file.sync_all() {
+            eprintln!("Error syncing tempfile: {e}");
+        }
+
+        if let Err(e) = remove_file(&self.temp_path) {
```

Comment:
I believe this will fail if we move the file (which happens in the success case).